File: javascript/matrixOperations.js
const math = require('mathjs');

function add(matrixA, matrixB) {
    return math.add(matrixA, matrixB);
}

function subtract(matrixA, matrixB) {
    return math.subtract(matrixA, matrixB);
}

function scalar_multiply(matrix, scalar) {
    return math.multiply(matrix, scalar);
}

function transpose(matrix) {
    return math.transpose(matrix);
}

function sum(matrix) {
    return math.sum(matrix);
}

function multiply(matrixA, matrixB) {
    return math.multiply(matrixA, matrixB);
}

function vectorDotProduct(vectorA, vectorB) {
    return math.dot(vectorA, vectorB);
}

function vectorCrossProduct(vectorA, vectorB) {
    return math.cross(vectorA, vectorB);
}

function inverse(matrix) {
    return math.inv(matrix);
}

module.exports = { 
    add,
    subtract,
    scalar_multiply,
    transpose,
    sum,
    multiply,
    vectorDotProduct,
    vectorCrossProduct,
    inverse
};

File: matlab/MatrixOperations.m
classdef MatrixOperations
    methods(Static)
        function result = add(matrixA,matrixB)
            result = matrixA + matrixB;
        end
        
        function result = subtract(matrixA,matrixB)
            result = matrixA - matrixB;
        end
        
        function result = scalarMultiply(matrix,scalar)
            result = matrix * scalar;
        end
        
        function result = transpose(matrix)
            result = transpose(matrix);
        end
        
        function result = sum(matrix)
            result = sum(sum(matrix));
        end
        
        function result = multiply(matrixA,matrixB)
            result = matrixA * matrixB;
        end
        
        function result = vectorDotProduct(vectorA,vectorB)
            result = dot(vectorA, vectorB);
        end
        
        function result = vectorCrossProduct(vectorA,vectorB)
            result = cross(vectorA, vectorB);
        end
        
        function result = inverse(matrix)
            result = inv(matrix);
        end
    end
end